cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+        args["task_type"] = "Batch"
+        args["task_type_parameters"] = \
+            [compilation_param,
+             [input_filename, output_filename],
+             evaluation_param]
+        public_groups = self.conf.get('public_groups', [0, 1])
+
+        args["testcases"] = {}
+        tests_per_group = [0] * len(points_per_group)
+        test_zip = os.path.join(self.task_dir, self.conf.get('test_archive', 'testi.zip'))
+        with zipfile.ZipFile(test_zip) as zip:
+
+            # Collect and organize test files from zip archive
+            matcher = re.compile(r"\.(i|o)(\d+)([a-z]*)$")
+            test_files = {}
+            for test_filename in zip.namelist():
+                if '/' in test_filename or '\\' in test_filename:
+                    logger.critical("Test zip archive contains a directory")
+                    return None
+                match = matcher.search(test_filename)
+                if not match:
+                    logger.critical(f"Unsupported file in test archive {name}")
+                    return None
+
+                is_input = match.group(1) == 'i'
+                group = int(match.group(2))
+                test_in_group = match.group(3)
+
+                if group not in test_files:
+                    test_files[group] = {}
+                if test_in_group not in test_files[group]:
+                    test_files[group][test_in_group] = {}
+
+                testcase = test_files[group][test_in_group]
+                testcase['input' if is_input else 'output'] = test_filename
+
+            # Extract them in correct order and update subtask list and testcases
+            for group in sorted(test_files.keys()):
+                for test_in_group in sorted(test_files[group].keys()):
+                    tests_per_group[group] += 1
+                    testcase = test_files[group][test_in_group]
+                    if 'input' not in testcase or 'output' not in testcase:
+                        logger.critical(f"Input or output file not found for test {group}{test_in_group}")
+                        return None
+                    with zip.open(testcase['input'], 'r') as input_file:
+                        content = io.TextIOWrapper(input_file, encoding='ascii', newline=None).read()
+                        input_digest = self.file_cacher.put_file_content(
+                            content.encode('ascii'),
+                            f"Input {testcase['input']} for task {task.name}")
+                    with zip.open(testcase['output'], 'r') as output_file:
+                        content = io.TextIOWrapper(output_file, encoding='ascii', newline=None).read()
+                        output_digest = self.file_cacher.put_file_content(
+                            content.encode('ascii'),
+                            f"Output {testcase['output']} for task {task.name}")
+                    codename = f"{group:03}{test_in_group}"
```

Comment:
What do you think about lowering this to two digits? I understand the intent, but it should be rare, and seeing 001 002 003 … in the testcase list makes it harder to read, at least for me.

Reply:
Modified it to automatically use the minimum necessary digit count.